src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteMefComposition.cs

```diff
@@ -13,25 +14,34 @@ internal static class RemoteMefComposition
 {
     internal static readonly ImmutableArray<Assembly> RemoteHostAssemblies = [typeof(RemoteMefComposition).Assembly];
 
-#pragma warning disable VSTHRD012 // Provide JoinableTaskFactory where allowed
-    private static readonly AsyncLazy<ExportProvider> s_exportProviderLazy = new(CreateExportProviderAsync);
-#pragma warning restore VSTHRD012 // Provide JoinableTaskFactory where allowed
+    private static readonly AsyncLazy<CompositionConfiguration> s_lazyConfiguration = new(CreateConfigurationAsync, joinableTaskFactory: null);
 
-    public static Task<ExportProvider> GetExportProviderAsync()
-        => s_exportProviderLazy.GetValueAsync();
+    private static readonly AsyncLazy<ExportProvider> s_lazyExportProvider = new(CreateExportProviderAsync, joinableTaskFactory: null);
 
-    // Internal for testing
-    // Inspired by https://github.com/dotnet/roslyn/blob/25aa74d725e801b8232dbb3e5abcda0fa72da8c5/src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs#L77
-    internal static async Task<ExportProvider> CreateExportProviderAsync()
+    public static Task<CompositionConfiguration> GetConfigurationAsync(CancellationToken cancellationToken = default)
```

Comment:
Doesn't seem to be a need for the `CancellationToken` here to be optional.

Additionally, though a nit, I would prefer it wasn't optional on the `GetExportProviderAsync` method, and just have the service factory base be explicit about passing in None/default.